defender/downloaded/Forta_Distributions.js

```diff
@@ -1,155 +1,101 @@
-/* eslint-disable import/no-unresolved,import/no-extraneous-dependencies */
-const ethers = require('ethers');
-const axios = require('axios');
-/* eslint-enable import/no-unresolved,import/no-extraneous-dependencies */
+// Set the name of the Secret set in Autotask
+const discordSecretName = 'SecurityAlertsDiscordUrl';
 
-const fortaApiEndpoint = 'https://api.forta.network/graphql';
+const axios = require('axios');
 
-// axios post request for forta graphql api
 async function post(url, method, headers, data) {
   return axios({
-    url, method, headers, data,
+    url,
+    method,
+    headers,
+    data,
   });
 }
 
-function createDiscordMessage(compAccrued, compDistributed, receiver, transactionHash) {
-  const receiverFormatted = receiver.slice(0, 6);
-
-  // // construct the Etherscan transaction link
-  const etherscanLink = `[TX](<https://etherscan.io/tx/${transactionHash}>)`;
-
-  // handle case for infinity % (no comp accrued before)
-  if (compAccrued.eq(0) || compDistributed.eq(0)) {
-    return `${etherscanLink} 🌊 **${receiverFormatted}** previously had no **COMP** accrued and was just distributed **COMP** tokens`;
-  }
-
-  // % distribution % is calculated by compDistributed / compAccrued * 100
-  // format the number to have comma separators for powers of 1000
-  const internationalNumberFormat = new Intl.NumberFormat('en-US');
-
-  let percentageMore = compDistributed.div(compAccrued).mul(100).toString();
-  percentageMore = internationalNumberFormat.format(percentageMore);
-
-  return `${etherscanLink} 🌊 **${percentageMore}%** more **COMP** distributed to **${receiverFormatted}** than expected`;
-}
-
-// post to discord
 async function postToDiscord(url, message) {
   const method = 'post';
   const headers = {
     'Content-Type': 'application/json',
   };
-  const data = { content: message };
+  const data = JSON.stringify({ content: message });
 
   let response;
   try {
     // perform the POST request
     response = await post(url, method, headers, data);
   } catch (error) {
-    // is this a "too many requests" error (HTTP status 429)
+    // check if this is a "too many requests" error (HTTP status 429)
     if (error.response && error.response.status === 429) {
       // the request was made and a response was received
-      // try again after waiting 5 seconds
+      // try again after waiting 5 - 50 seconds, if retry_after value is received, use that.
```

Comment:
Check out axios-retry for future projects.
https://github.com/softonic/axios-retry